bitnami/influxdb/templates/influxdb/configmap-backup.yaml

```diff
@@ -60,19 +60,59 @@ data:
         done
 
         echo "deleting old backups"
-        find {{ .Values.backup.directory }} -mindepth 3 -maxdepth 3 -not -name ".snapshot" -not -name "lost+found" -type d -mtime +{{ .Values.backup.retentionDays }} -exec rm -r {} \;
+        find {{ .Values.backup.directory.local }} -mindepth 3 -maxdepth 3 -not -name ".snapshot" -not -name "lost+found" -type d -mtime +{{ .Values.backup.retentionDays }} -exec rm -r {} \;
     fi
-  upload-google.sh: |-
+  upload.sh: |-
     #!/bin/sh
 
     set -e
 
-    gcloud auth activate-service-account --key-file /var/secrets/google/{{ .Values.backup.uploadProviders.google.secretKey }}
-    gsutil -m rsync -r -d {{ .Values.backup.directory }}/ {{ .Values.backup.uploadProviders.google.bucketName }}
-  upload-azure.sh: |-
-    #!/bin/sh
+    # Google Cloud Storage
+    if [[ {{ .Values.backup.uploadProviders.googlecloudstorage.enabled }} == "true" ]]; then
```

Comment:
I would generate the proper bash script only if the google cloud storage is enabled, so instead of adding that check in the if instruction I would just render the bash code for Google only if the value is true

Reply:
I was trying to avoid mounting different scripts for different providers to keep the overall code clean and easy to handle. Having a single script with all the checks does exactly. If you notice, I have added other provider related code in the same script.

Reply:
The recommendation by @javsalgar doesn't propose using different scripts, but rendering different bash code blocks depending on the cloud provider. The resulting script that will be added to the ConfigMap will be simpler using this approach.